### ISSUE TITLE
C.121: "Reason" part tries to explain why containing data in interface class is bad, but example fails to show it

### ISSUE THREAD

**Issue by NooneAtAll** [isocpp/CppCoreGuidelines#1535]
[C.121](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-abstract) has  
```A class is more stable (less brittle) if it does not contain data.```
as its first sentence, but its example only shows why virtual destructor is needed

Why is it bad to have data members in interfaces?

**Comment by shaneasd** [comment]
I agree. The example in C.121 seems like it's better suited to illustrating [C.35](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c35-a-base-class-destructor-should-be-either-public-and-virtual-or-protected-and-nonvirtual) and personally I think the existing example in C.35, unsurprisingly, does a better job of it.
